go/src/koding/klient/machine/machinegroup/mount.go

```diff
@@ -141,3 +142,118 @@ func (g *Group) AddMount(req *AddMountRequest) (res *AddMountResponse, err error
 		MountID: mountID,
 	}, nil
 }
+
+// ListMountRequest defines machine group mount list request.
+type ListMountRequest struct {
+	// ID is an optional identifier for the remote machine. If set, only
+	// mounts related to this machine will be returned.
+	ID machine.ID `json:"id,omitempty"`
+
+	// MountID is an optional identifier of a mount which is meant to be listed.
+	MountID mount.ID `json:"mountID,omitempty"`
+}
+
+// ListMountResponse defines machine group mount list response.
+type ListMountResponse struct {
+	// Mounts is a map that contains machine aliases as keys and created mount
+	// infos which store the current synchronization status of mount.
+	Mounts map[string][]sync.Info `json:"mounts"`
+}
+
+// ListMount checks the status of mounts and returns their infos. This function
+// guarantees that if returned error is nil, response Mounts field is always
+// non-nil. It doesn't fail if provided filters contain incorrect values.
+func (g *Group) ListMount(req *ListMountRequest) (*ListMountResponse, error) {
```

Comment:
could you separate this function into smaller logical parts?, it is hard ( for me ) to follow it